src/main/scala/devices/tilelink/PhysicalFilter.scala

```diff
@@ -23,53 +25,103 @@ class DevicePMP(params: DevicePMPParams) extends GenericParameterizedBundle(para
-      field(params.addressBits-params.pageBits, addr_hi, l(0) || blockAddress),
+      field(params.addressBits-params.pageBits, addr_hi, addrHiDesc,
+        l(0) || blockAddress),
       RegField(56 - (params.addressBits-2)),
-      field(1, r),
-      field(1, w),
+      field(1, r, rDesc),
+      field(1, w, wDesc),
       RegField(1), // x
-      field(1, a),
+      field(1, a, aDesc),
       RegField(3), // a high + 2 reserved
-      field(1, l))
+      field(1, l, lDesc))
   }
 }
 
+
+/* Initial value of the PMP registers
+ *  
+ *  @param address - Initial value of the address field. Will be shifted down by pageBits before being stored in the PMP register addr_hi register.
+ *  @param l Initial value of the l (locked) bit. True means no modifications of other fields are allowed.
+ *  @param a Initial value of the a (address match mode) bit. False means disabled, True means use TOR address matching. Also disables modification of previous PMP address.
+ *  @param r Initial value of the r (read access match) bit. True means allow read accesses.
+ *  @param w Initial value of the w (write access match) bit. True means allow write accesses.
+ *  
+ */
 case class PMPInitialValue(address: BigInt = 0, l: Boolean = false, a: Boolean = false, r: Boolean = false, w: Boolean = false)
+case class PMPInitialValueInt(addr_hi: BigInt = 0, l: Int, a: Int, r: Int, w: Int)
 
 object DevicePMP
 {
-  def apply(addressBits: Int, pageBits: Int, initial: Option[PMPInitialValue] = None) = {
+  /** Create DevicePMP bundle with appropriate initial value intended for use in RegInit
+    *  
+    *  @param addressBits width of addresses
+    *  @param pageBits number of address bits per page
+    *  @param initial optional initial value. 0 is used for all fields if not specified.
+    *  
+    */
+
+  def apply(addressBits: Int, pageBits: Int, initial: Option[PMPInitialValue] = None): DevicePMP = {
+
     val out = Wire(new DevicePMP(DevicePMPParams(addressBits, pageBits)))
+    val outInts = getInitialValueInts(addressBits, pageBits, initial)
 
-    initial.foreach { i =>
-      require ((i.address >> addressBits) == 0)
-      require ((i.address >> pageBits) << pageBits == i.address)
-      out.addr_hi := UInt(i.address >> pageBits)
-    }
+    out.addr_hi := outInts.addr_hi.U
+    out.l := outInts.l.U
+    out.a := outInts.a.U
+    out.r := outInts.r.U
+    out.w := outInts.w.U
 
-    def get(f: PMPInitialValue => Boolean) = initial.map(x => Bool(f(x)).asUInt).getOrElse(UInt(0))
-    out.l := get(_.l)
-    out.a := get(_.a)
-    out.r := get(_.r)
-    out.w := get(_.w)
     out
   }
-}
 
-/** PhysicalFilter uses a set of DevicePMP registers to control whether
-  * accesses of certain types are allowed to proceed or denied. The Filter
-  * will only prevent acquisition of NEW permissions; it will not shoot
-  * down permissions Acquired previously.
-  */
+  /** Helper to convert from config booleans to integer values needed for the DevicePMP Bundles */
```

Comment:
> from config booleans

but takes `Int`s and an `Option[PMPInitialValue]` as an argument?

Reply:
inside the PMPInitial value are Booleans and I want Ints.... will clarify